tests/unit-tests/test-class-lesson.php

```diff
@@ -26,11 +96,39 @@ public function setup() {
 
 		$this->factory = new Sensei_Factory();
 		Sensei_Test_Events::reset();
+
+		global $current_screen, $taxnow, $typenow, $post, $current_user, $wp_scripts;
+		$this->initial_screen     = $current_screen;
+		$this->initial_taxnow     = $taxnow;
+		$this->initial_typenow    = $typenow;
+		$this->initial_post       = $post;
+		$this->initial_user       = $current_user;
+		$this->initial_wp_scripts = $wp_scripts;
+
+		$this->initial_course   = Sensei()->course;
+		$this->initial_lesson   = Sensei()->lesson;
+		$this->initail_quiz     = Sensei()->quiz;
+		$this->initial_question = Sensei()->question;
+		$this->initial_notices  = Sensei()->notices;
 	}
 
 	public function tearDown() {
 		parent::tearDown();
 		$this->factory->tearDown();
+
+		global $current_screen, $taxnow, $typenow, $post, $current_user, $wp_scripts;
+		$current_screen = $this->initial_screen;
+		$taxnow         = $this->initial_taxnow;
+		$typenow        = $this->initial_typenow;
+		$post           = $this->initial_post;
+		$current_user   = $this->initial_user;
+		$wp_scripts     = $this->initial_wp_scripts;
```

Comment:
I think resetting the WordPress globals is not needed. It's done in the [parent tear down method](https://github.com/WordPress/wordpress-develop/blob/trunk/tests/phpunit/includes/abstract-testcase.php#L155-L181).

Reply:
Unfortunately, it doesn't work as expected. Some of those variable affected other tests :( I added restoration of them only as a reaction to those failures ([example 1](https://github.com/Automattic/sensei/runs/7556457000?check_suite_focus=true), [example 2](https://github.com/Automattic/sensei/runs/7557354642?check_suite_focus=true)).

Reply:
Hm, strange. If I comment out these lines locally all tests still are passing. Could you double-check? Maybe it only fails in CI? If that's the case, I'm ok merging it as is. 🙂

Reply:
Sure, I'll check it. I could reproduce those issues locally. 
By the way, not all the variables caused errors and failures, but I restore all of them for consistency.

Reply:
Looks you were right, and it works without restoring global variables. Probably confused with Sensei globals.

Reply:
Haha, look at this: https://github.com/Automattic/sensei/runs/7639272707?check_suite_focus=true

Reply:
Returned restoration of current screen and related global variables.
Tests fail without them for PHP7.2 and WP 5.8, WP 5.7.

Reply:
Cool, I'm good with that. 🙂